apps/content_filters/content_filter/content_filter_service.py

```diff
@@ -230,18 +232,22 @@ def _does_filter_condition_match(self, content_filter, article, filters, express
                     logger.error("Missing filter condition %s in content filter %s", f, content_filter.get("name"))
                     return False
                 filter_condition = FilterCondition.parse(fc)
+                if not cache:
+                    return filter_condition.does_match(article)
                 setattr(flask.g, cache_id, filter_condition.does_match(article))
             if not getattr(flask.g, cache_id):
                 return False
         return True
 
-    def _does_content_filter_match(self, content_filter, article, filters, expression) -> bool:
+    def _does_content_filter_match(self, content_filter, article, filters, expression, cache=True) -> bool:
         for f in expression["expression"]["pf"]:
             cache_id = _cache_id("content-filter-match", f, article)
-            if not hasattr(flask.g, cache_id):
+            if not cache or not hasattr(flask.g, cache_id):
                 current_filter = (
                     filters.get("content_filters", {}).get(f, {}).get("cf") if filters else self.get_cached_by_id(f)
                 )
+                if not cache:
+                    return self.does_match(current_filter, article, filters=filters)
```

Comment:
I think here you have to pass the `cache` param

Reply:
my bad I missed this